src/snowflake/cli/api/project/schemas/updatable_model.py

```diff
@@ -14,17 +14,125 @@
 
-from pydantic import BaseModel, ConfigDict, Field
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    ValidationInfo,
+    field_validator,
+)
+from pydantic.fields import FieldInfo
 from snowflake.cli.api.project.util import IDENTIFIER_NO_LENGTH
 
+PROJECT_TEMPLATE_START = "<%"
+
+
+def _is_templated(info: ValidationInfo, value: Any) -> bool:
+    return (
+        info.context
+        and info.context.get("includes_templates", False)
+        and isinstance(value, str)
+        and PROJECT_TEMPLATE_START in value
+    )
+
+
+def field_validator_allowing_templates(*validator_args, **validator_kwargs):
+    """
+    This validator replaces field_validator from Pydantic.
+    The difference is that this validator will skip validation
+    whenever a value is a String and the value is templated.
+
+    It also checks the context to ensure includes_templates is set to True.
+    Otherwise, if includes_templates context is not set, it behaves the same
+    way as field_validator, and does not check templates anymore.
+    """
+    mode = validator_kwargs.get("mode", "after")
+
+    def decorator(func):
+        def wrapper_mode_wrap(cls, value, handler, info: ValidationInfo, **kwargs):
+            if _is_templated(info, value):
+                return value
+            return func(cls, value, handler, **kwargs)
+
+        def wrapper_default(cls, value, info: ValidationInfo, **kwargs):
+            if _is_templated(info, value):
+                return value
+            return func(cls, value, **kwargs)
+
+        wrapper = wrapper_mode_wrap if mode == "wrap" else wrapper_default
+        return field_validator(*validator_args, **validator_kwargs)(
+            classmethod(wrapper)
+        )
+
+    return decorator
+
+
+_initial_context: ContextVar[Optional[Dict[str, Any]]] = ContextVar(
+    "_init_context_var", default=None
+)
+
+
+@contextmanager
+def context(value: Dict[str, Any]) -> Iterator[None]:
+    """
+    Thread safe context for Pydantic.
+    By using `with context()`, you ensure context changes apply
+    to the with block only
+    """
+    token = _initial_context.set(value)
+    try:
+        yield
+    finally:
+        _initial_context.reset(token)
+
 
 class UpdatableModel(BaseModel):
     model_config = ConfigDict(validate_assignment=True, extra="forbid")
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(**kwargs)
+    def __init__(self, /, **data: Any) -> None:
+        self.__pydantic_validator__.validate_python(
+            data,
+            self_instance=self,
+            context=_initial_context.get(),
+        )
+
+    @classmethod
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+
+        # Collect all the Pydantic fields
+        # from all the classes in the inheritance chain
+        all_fields = {}
+        for class_ in cls.__mro__:
+            if hasattr(class_, "__annotations__"):
```

Comment:
It's less forgiving in terms of class orders because it evaluates the types. Not a big deal, but I need to loop over the hierarchy to collect field values as well, so I kept it this way - please check updated version.